### DISCUSSION TITLE
Feature idea: More intuitive/powerful use of Faikin auto mode via HA/MQTT

### DISCUSSION THREAD

**Discussion by timfrazee** [revk/xESP32-Faikout#309] [Ideas]
I have 3 Faikin units, all working fantastically. They all run Faikin auto mode with `autop` on, meaning they turn off when within the desired temp range for a while.

A minor but irritating issue is that the Faikin reports the unit as off to HA/MQTT when it's off due to `autop`. This may be technically accurate, but it hides the fact that the unit is still running Faikin auto mode and may turn itself back on without user input. In some interfaces (such as Homekit), a thermostat in "Off" mode cannot have its target temp changed, so a user is required to manually turn the unit on to change the Faikin auto mode target temp.

Below is a suggestion to provide an alternate set of behaviors leveraging existing Faikin auto mode (hereafter simply referred to as `auto`) that will make control via HA (and other smart home apps) more intuitive.

The end goal is to have Faikin utilize the strengths of `auto` (such as using an external temp reference, allowing a customizable target temp range, etc) while maintaining expected, intuitive behaviors from thermostat controls typically found in smart home apps.

This presumably would be optional, and could be enabled with a boolean setting. Let's call that setting `autoalternate`, and assume from here on that this setting - as well as `auto`, unless otherwise specified - is enabled.

### New settings

- `autoheatonly` (bool) - When enabled, HVAC mode changes to "Heat". `auto` would behave as normal, except that it would not switch into cooling mode even if the room temp exceeds the target.
- `autocoolonly` (bool) - When enabled, HVAC mode changes to "Cool". `auto` would behave as normal, except that it would not switch into heating mode even if the room temp falls below the target.

Note: Only one of these settings can be on at a time. Toggling one on turns the other off.

## New behaviors

- When the user changes the HVAC mode to "Auto", Faikin turns `autoheatonly` and `autocoolonly` off (existing normal `auto` behavior).
- When the user changes the HVAC mode to "Heat", Faikin turns `autoheatonly` on.
- When the user changes the HVAC mode to "Cool", Faikin turns `autocoolonly` on.
- When the user changes the HVAC mode to "Off", Faikin turns the HVAC unit off and sets `auto` to off (preventing Faikin from turning the HVAC unit back on without manual input).

Inversely:
- When `autoheatonly` is off, and `autocoolonly` is off: Faikin reports HVAC mode as "Auto"
- When `autoheatonly` is on: Faikin reports HVAC mode as "Heat"
- When `autocoolonly` is on: Faikin reports HVAC mode as "Cool"
- The above are all true even if the HVAC unit is off due to `autop`
- When `auto` is off and the HVAC unit is off: Faikin reports HVAC mode as "Off"

### Thoughts

I've only given a cursory glance at the source code, but I'd be happy to see if I can contribute to this idea. The suggestion, of course, is just one way this concept could be implemented.

Lastly, thank you revk for Faikin. It's done a lot to take Daikin mini split units from passable to exceptional.

**Comment by revk** [comment]
I have a setting not to switch heat/cold. I assume that is not quite what you are after?

**Comment by revk** [comment]
I also wish I had never added hvac mode!

**Comment by timfrazee** [comment]
@revk Sorry for the (very) late reply.

In short, what I want is manual control over the HVAC mode while in Faikin auto mode. In other words, if I go into HA and set the mode to "Heat", I want the unit to only heat, "Cool" to only cool, "Off" to be off, and "Auto" to be current Faikin auto behavior. Importantly, I want this behavior to stick until it's changed.

To explain in a slightly different manner (with Faikin auto mode on at all times):

**Current Behavior:**

If I manually change the HVAC mode to anything other than Auto, Faikin auto mode eventually changes it back to Auto, and will both heat and cool.

- If I turn the unit off, it will eventually turn itself back on.
- If I set the unit to Heat, Cool, or Fan, it will eventually set itself back to Auto.

**Desired Behavior:**

If I manually change the HVAC mode to anything other than Auto, Faikin respects the selection while otherwise maintaining its behavior.

- If I turn the unit off (or fan), the unit should remain off (or fan) until the mode is changed.
- If I set the unit to Heat or Cool, it will only heat or cool, respectively, until the mode is changed.
  - Faikin will use auto-mode behaviors while in the requested mode, but otherwise just idle (or turn off if `autop` is set).

**Use Case:**

This would be helpful, for example, during the summer where I don't want to ever heat the house, even if the temp drops at night. Likewise, during the winter when I don't want to ever cool the house, even if the temp rises at midday.

Additionally, if I open a door or window, I want to be able to turn the unit off (or to fan mode) and have it not turn back on until I tell it to.

All, ideally, while keeping Faikin auto mode on, because I greatly appreciate auto mode's use of a BLE sensor to maintain real-world room temps.

**Reply by revk** [reply]
Don't the settings allow this? There is some sort of mode switch disallow in settings already. I can look more in morning, but I am sure we have loads of controls to cover this.

**Reply by timfrazee** [reply]
I didn't see it, but I very well may have missed it. I'll take a look as well, and close this out if it is indeed possible.

Thanks!

**Reply by revk** [reply]
There are settings to stop its auto turning on and off and settings to stop is switching heat/cool mode.

**Comment by cmonty14** [comment]
Hi, is there any update on this topic?
